src/rcl_logging_syslog.cpp

```diff
@@ -32,12 +33,92 @@
 #include <rcutils/strdup.h>
 #include <rcutils/time.h>
 
+static const char * facility_env_name = "RCL_LOGGING_SYSLOG_FACILITY";
+
 // see https://codebrowser.dev/glibc/glibc/misc/syslog.c.html#LogTag
 // This memory needs to be kept until closelog()
 static std::shared_ptr<std::string> syslog_identity;
 
 static const char *logger_name = "rcl_logging_syslog";
 
+typedef struct facility_index {
+  const char *c_name;
+  const int c_value;
+} FACILITY_INDEX;
+
+const FACILITY_INDEX facility_table[] =
+  {
+    { "LOG_CRON", LOG_CRON },
+    { "LOG_DAEMON", LOG_DAEMON },
+    { "LOG_SYSLOG", LOG_SYSLOG },
+    { "LOG_USER", LOG_USER },
+    { "LOG_LOCAL0", LOG_LOCAL0 },
+    { "LOG_LOCAL1", LOG_LOCAL1 },
+    { "LOG_LOCAL2", LOG_LOCAL2 },
+    { "LOG_LOCAL3", LOG_LOCAL3 },
+    { "LOG_LOCAL4", LOG_LOCAL4 },
+    { "LOG_LOCAL5", LOG_LOCAL5 },
+    { "LOG_LOCAL6", LOG_LOCAL6 },
+    { "LOG_LOCAL7", LOG_LOCAL7 },
```

Comment:
other facilities are meant to be specific facility such as ftp, kernel and so on. so here just added the generic user space facilities only.